contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -596,10 +607,42 @@ add_query_hints(PLtsql_expr *expr)
 	if (!leading_hint.empty())
 		hint += leading_hint;
 	hint += "*/";
+	return hint;
+}
+
+static char *
+format_query_with_hints(char *query, std::string hints)
+{
 	StringInfoData new_query;
 	initStringInfo(&new_query);
-	appendStringInfo(&new_query, "%s %s", const_cast <char *>(hint.c_str()), expr->query);
-	expr->query = new_query.data;
+	int initialTokenEnd = find_hint_offset(query);
+
+	appendStringInfo(&new_query, "%s", psprintf("%.*s", initialTokenEnd, query));
+	appendStringInfo(&new_query, " %s", const_cast <char *>(hints.c_str()));
+	appendStringInfo(&new_query, " %s", &query[initialTokenEnd]);
+	return new_query.data;
+}
+
+static int
+find_hint_offset(char *query)
+{
+	std::string queryString(query);
+	size_t spaceIdx = queryString.find_first_of(" \t\r\n");
+	size_t commentStartIdx = queryString.find("/*");
+
+	//if there is no space and no comment default to beginning of statement
+	if (commentStartIdx == std::string::npos && spaceIdx == std::string::npos)
+		return 0;
+
+	//if both comments and space return the index of the smaller.
+	if (commentStartIdx != std::string::npos && spaceIdx != std::string::npos)
+		return min(static_cast<int>(spaceIdx), static_cast<int>(commentStartIdx));
+
+	//if no comment return spaceIdx
+	if (commentStartIdx == std::string::npos)
+		return static_cast<int>(spaceIdx);
+
+	return 0;
```

Comment:
Do you have a test case for this?

Reply:
This should not be a code path that should be reachable.  I have tested it locally with a query using tabs before when we were searching for a single ' ' character and it worked as one would expect.  Since this is not a code path that should be normally reachable, I don't know how to test for it with a scripted test.